geth/node/rpc.go

```diff
@@ -60,27 +64,39 @@ func NewRPCManager(nodeManager common.NodeManager) *RPCManager {
 
 // Call executes RPC request on node's in-proc RPC server
 func (c *RPCManager) Call(inputJSON string) string {
-	server, err := c.nodeManager.RPCServer()
+	config, err := c.nodeManager.NodeConfig()
 	if err != nil {
 		return c.makeJSONErrorResponse(err)
 	}
 
 	// allow HTTP requests to block w/o
 	outputJSON := make(chan string, 1)
 	go func() {
-		httpReq := httptest.NewRequest("POST", "/", strings.NewReader(inputJSON))
-		rr := httptest.NewRecorder()
-		server.ServeHTTP(rr, httpReq)
-
-		// Check the status code is what we expect.
-		if respStatus := rr.Code; respStatus != http.StatusOK {
-			log.Error("handler returned wrong status code", "got", respStatus, "want", http.StatusOK)
-			outputJSON <- c.makeJSONErrorResponse(ErrRPCServerCallFailed)
+		body := bytes.NewBufferString(inputJSON)
+
+		if !config.UpstreamConfig.Enabled {
```

Comment:
I think it would be possible to refactor this part of code a little bit to avoid duplication:

```go
var resbody []byte
var err error

if !config.UpstreamConfig.Enabled {
  resbody, err = c.callNodeStream(body)
} else {
  resbody, err = c.callUpstreamStream(config, body)
}

if err != nil {
  outputJSON <- c.makeJSONErrorResponse(err)
  return
}

outputJSON <- string(resbody)
```

Feel free to skip this as this is refactoring.